---
title: "Introduction_1_loadvariants"
author: "Hangjia Zhao"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{introduction_I_loadvariants}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Load library 

```{r setup}
library(pgxRpi)
```

## `pgxLoader` function in this package

This function loads various data from `Progenetix` database.  

The parameters of this function:

* `type` A string specifying output data type. Available options are "biosample", "individual",
"variant" or "frequency". 
* `output` A string specifying output file format. When the parameter `type` is "variant",
available options are NULL, "pgxseg" ,"pgxmatrix", "coverage" or "seg" ; When the parameter `type` is "frequency", available options are "pgxfreq" or "pgxmatrix" .
* `filters` A single or a comma-concatenated list of identifiers for cancer type,
literature, and cohorts such as c("NCIT:C7376","pgx:icdom-98353","PMID<!-- -->:22824167", "pgx:cohort-TCGAcancers").
* `codematches` A logical value determining whether to exclude samples 
from child concepts of specified filters that belong to cancer type/tissue encoding system (NCIt, icdom/t, Uberon). 
If TRUE, retrieved samples only keep samples exactly encoded by specified filters. 
Don't use this parameter when `filters` include cancer-irrelevant identifiers such as PMID and cohort filters.
Default is FALSE.
* `limit` Integer to specify the number of returned samples/individuals/coverage profiles for each filter. 
Default is 0 (return all). 
* `skip` Integer to specify the number of skipped samples/individuals/coverage profiles for each filter. 
E.g. if skip = 2, limit=500, the first 2*500 =1000 profiles are skipped and the next 500 profiles are returned. 
Default is NULL (no skip). 
* `biosample_id` A single or a comma-concatenated list of identifiers used in Progenetix 
database for identifying biosamples. 
* `individual_id` A single or a comma-concatenated list of identifiers used in Progenetix 
database for identifying individuals. 
* `save_file` A logical value determining whether to save the variant data as file 
instead of direct return. Only used when the parameter `type` is "variant" and `output` is "pgxseg" or "seg". Default is FALSE.
* `filename` A string specifying the path and name of the file to be saved. 
Only used if the parameter `save_file` is TRUE. Default is "variants.seg/pgxseg" in current work directory.

# Retrieve meatdata of samples from specific groups of interests

Relevant parameters of `pgxLoader` function: type, filters, skip, limit,codematches,biosample_id, individual_id

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3512")
# data looks like this
biosamples[c(1700:1705),]
```

The data contains many columns representing different aspects of sample information. 
By default, it returns all related samples (limit=0). You can access a subset of them 
via the parameter `limit` and `skip`. For example, if you want to access the first 1000 samples
, you can set `limit` = 1000, `skip` = 0. Some samples may have multiple records because they
are analysed more than one time and have different analysis ids (callset_ids).

```{r}
biosamples_2 <- pgxLoader(type="biosample", filters = "NCIT:C3512",skip=0, limit = 1000)
# Dimension: Number of samples * features
print(dim(biosamples))
print(dim(biosamples_2))
```

The number of samples in specific group can be queried by  `pgxCount` function.

```{r}
pgxCount(filters = "NCIT:C3512")
```

The NCIt code of retrieved samples doesn't only contain specified filters but contains child terms. 

```{r}
unique(biosamples$histological_diagnosis_id)
```

Setting `codematches` as TRUE allows this function to only return biosamples with exact match to the filter.

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3512",codematches = TRUE)

unique(biosamples$histological_diagnosis_id)
```

You can also retrieve biosamples by biosample id and corresponding individual id 

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3512",codematches = TRUE,
                        biosample_id = c("pgxbs-kftva5zv","pgxbs-kftva5zw"),
                        individual_id = "pgxind-kftx26ml")
```

The result only contains samples with the specified filter NCIT:C3512, biosample id and individual id

```{r}
biosamples[biosamples$histological_diagnosis_id != "NCIT:C3512",]
```

# Retrieve meatdata of individuals from specific groups of interests 

Relevant parameters of `pgxLoader` function: type, filters, skip, limit,codematches,individual_id

If you want to query metadata (e.g. survival data) of individuals where the samples 
of interest come from, you can either search by filters:

```{r}
individuals <- pgxLoader(type='individual',filters='NCIT:C3270',codematches = T)
# Dimension: Number of individuals * features
print(dim(individuals))
# data looks like this
individuals[c(36:40),]
```
or by individual id which you can get from the query of sample

```{r}
individual <- pgxLoader(type='individual',individual_id = 'pgxind-kftx26ml')

individual
```

or by both

```{r}
total.individuals <- pgxLoader(type='individual',filters='NCIT:C3270',codematches=T,
                               individual_id = 'pgxind-kftx26ml')

dim(total.individuals)
```


# Retrive CNV coverage of biosamples from specific groups of interests

## Across genomic bins 

Relevant parameters of `pgxLoader` function: type, output, filters, codematches, skip,limit

```{r}
cnv.status <- pgxLoader(type="variant", output="pgxmatrix", filters = "NCIT:C9245")
```

By default, it returns all available profiles (limit=0), so the query may spend several minutes 
when the number of retrieved samples is large. It is noted that the number of analysis 
profiles does not necessarily equal the number of samples.

The data looks like this

```{r}
print(dim(cnv.status))
cnv.status[c(1:3), c(1:5,6213:6215)]
```

In this dataframe, `analysis_id` is the identifier for individual analysis, `biosample_id` is the identifier for
individual biosample. One biosample_id may correspond to multiple analysis_id. `group_id` equals the meaning of `filters`. It's followed by all “gain status” columns (3106 intervals) plus all “loss status” columns (3106 intervals). The status is indicated by a coverage value, i.e. the fraction of how much the binned interval overlaps with one or more CNVs of the given type (DUP/DEL). For example, if the column `X1.400000.1400000.DUP` is 0.200 in one row, it means that one or more duplication events overlapped with 20% of the genomic bin located in chromosome 1: 400000-1400000 in the corresponding analysis.

Setting `codematches = True` can exclude profiles with group_id belonging to child terms of the input filters.

```{r}
cnv.status <- pgxLoader(type="variant", output="pgxmatrix", filters = "NCIT:C9245",
                        codematches = T)
```

Only 2745 profiles are left.

```{r}
dim(cnv.status)
```

## Across chromosomes or the whole genome 

Relevant parameters of `pgxLoader` function: type, output, filters, codematches, skip, limit

```{r}
cnv_covergae <- pgxLoader(type="variant", output="coverage", filters = "NCIT:C2948")
```

It includes CNV coverage across chromosome arms, whole chromosomes, or whole genome.

```{r}
names(cnv_covergae)
```

The data of CNV coverage across chromosomal arms looks like this

```{r}
head(cnv_covergae$chrom_arm_coverage)[,c(1:4, 49:52)]
```

The row names are id of biosamples from the group NCIT:C2948. There are 96 columns. 
The first 48 columns are duplication coverage across chromosomal arms, followed by deletion coverage.
The data of CNV coverage across whole chromosomes is similar, with the only difference in columns.

The data of CNV coverage across genome (hg38) looks like this

```{r}
head(cnv_covergae$whole_genome_coverage)
```

The first column is the total called coverage, followed by duplication coverage and deletion coverage.

Setting `codematches = True` can exclude profiles with group_id belonging to child terms of the input filters.
21 samples are excluded from the original 47 samples in this case.

```{r}
cnv_covergae_2 <- pgxLoader(type="variant", output="coverage", filters = "NCIT:C2948",
                            codematches = T)

print(dim(cnv_covergae$chrom_arm_coverage))
print(dim(cnv_covergae_2$chrom_arm_coverage))
```

# Retrieve segment variants by biosampleid

Relevant parameters of `pgxLoader` function: type, output, biosample_id, save_file, filename

Because of the time-out problem, the segment variant data can only be accessed by 
individual biosample id instead of filters.

## Get biosamples id from previously retrieved biosample information

```{r}
biosample_id <- biosamples$biosample_id[c(1:20)]
```

There are three output formats. 

## The first output format (by default)

 This format contains variant id and associated biosample id as well as analysis id. The variant is represented as 'DUP' (duplication) or 'DEL' (deletion) in specific chromosome locations. 

```{r}
variant_1 <- pgxLoader(type="variant", biosample_id = biosample_id)
head(variant_1)
```

## The second output format (`output` = "pgxseg")

This format is '.pgxseg' file format. It contains segment mean values (in `log2` column), which are equal to log2(copy number of measured sample/copy number of control sample (usually 2)). A few variants are point mutations represented by columns `reference_bases` and `alternate_bases`.

```{r}
variant_2 <- pgxLoader(type="variant", biosample_id = biosample_id,output = "pgxseg")
head(variant_2)
```

## The third output format (`output` = "seg")

This format is similar to the general '.seg' file format and compatible with IGV tool for visualization. The only difference between this file format and the general '.seg' file format is the fifth column. It represents variant type in this format while in the general '.seg' file format, it represents number of probes or bins covered by the segment. In addition, the point mutation variants are excluded in this file format.

```{r}
variant_3 <- pgxLoader(type="variant", biosample_id = biosample_id,output = "seg")
head(variant_3)
```

# Export variants data for visualization

Setting `save_file` as TRUE in `pgxLoader` function would make this function doesn't return 
variants data directly but let the retrieved data saved in the current work directory by default or other 
paths (specified by `filename`). The export is only available for variants data (type='variant'). 

## Upload 'pgxseg' file to Progenetix website

The following command creates a '.pgxseg' file with the name "variants.pgxseg" in "~/Downloads/" folder. 

```{r}
pgxLoader(type='variant', output='pgxseg', biosample_id=biosample_id, save_file=TRUE, 
          filename="~/Downloads/variants.pgxseg")
```

You can upload this '.pgxseg' file to [this link](https://progenetix.org/service-collection/uploader) 
for visualization.

## Upload '.seg' file to IGV

The following command creates a special '.seg' file with the name "variants.seg" in "~/Downloads/" folder. 

```{r}
pgxLoader(type='variant', output='seg', biosample_id=biosample_id, save_file=TRUE, 
          filename="~/Downloads/variants.seg")
```

You can upload this '.seg' file to [IGV tool](https://software.broadinstitute.org/software/igv/) for visualization.

# Session Info

```{r echo = FALSE}
sessionInfo()
```
