---
title: "Introduction_1_loadvariants"
author: "Hangjia Zhao"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{introduction_I_loadvariants}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Load library 

```{r setup}
library(pgxRpi)
```

## `pgxLoader` function in this package

This function loads various data from `Progenetix` database.  

The parameters of this function:

* `type` A string specifying output data type. Available options are "biosample", 
"variant" or "frequency". 
* `output` A string specifying output file format. When the parameter `type` is "variant",
available options are NULL, "pgxseg" ,"pgxmatrix", "coverage" or "seg" ; When the parameter `type` is "frequency",
available options are "pgxseg" or "pgxmatrix" .
* `filters` A single or a comma-concatenated list of identifiers for cancer type,
literature, and cohorts such as c("NCIT:C7376","pgx:icdom-98353","PMID:22824167", "pgx:cohort-TCGAcancers").
Only used if the parameter `type` is "biosample" or "frequency".
* `codematches` A logical value determining whether to exclude samples 
from child terms of specified filters. If FALSE, retrieved samples include child 
terms of specified filters. Only used when `filters` is specified. Default is FALSE.
* `biosample_id` A single or a comma-concatenated list of identifiers used in Progenetix 
database for identifying biosamples. 
* `limit` Integer to specify the number of returned samples/profiles. 
* `skip` Integer to specify the number of skipped samples/profiles. If skip = 2, limit=500, 
the first 2*500 =1000 profiles are skipped and the next 500 profiles are returned. 
* `save_file` A logical value determining whether to save the variant data as file 
instead of direct return. Only used when the parameter `type` is "variant" and `output` is "pgxseg" or "seg". Default is FALSE.
* `filename` A string specifying the path and name of the file to be saved. 
Only used if the parameter `save_file` is TRUE. Default is "variants.seg/pgxseg" in current work directory.

# Retrieve meatdata of samples from specific groups of interests

Relevant parameters of `pgxLoader` function: type, filters, codematches, biosample_id, skip, limit

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3059")
```

The data contains many columns representing different aspects of sample information. 
By default, it returns all related samples (limit=0). You can access a subset of them 
via the parameter `limit` and `skip`. For example, if you want to access the first 1000 samples
, you can set `limit` = 1000, `skip` = 0. 

```{r}
biosamples_2 <- pgxLoader(type="biosample", filters = "NCIT:C3059",skip=0, limit = 1000)

print(dim(biosamples))
print(dim(biosamples_2))
```

The number of samples in specific group can be queried by  `pgxCount` function.

```{r}
pgxCount(filters = "NCIT:C3059")
```

The NCIt code of retrieved samples doesn't only contain specified filters but contains child terms. 

```{r}
unique(biosamples$histological_diagnosis__id)
```

Setting `codematches` as TRUE allows this function to only return biosamples with exact match to the filter.

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3059",codematches = TRUE)

unique(biosamples$histological_diagnosis__id)
```

You can also retrieve biosamples by biosample id used in Progenetix database

```{r}
biosamples <- pgxLoader(type="biosample", filters = "NCIT:C3059",codematches = TRUE,
                        biosample_id = c("pgxbs-kftva5zv","pgxbs-kftva5zw"))
```

The result only contains samples with the specified filter NCIT:C3059 and the specified biosample id 

```{r}
unique(biosamples[biosamples$histological_diagnosis__id != "NCIT:C3059",'biosample_id'])
```

# Retrive CNV coverage of biosamples from specific groups of interests

## Across genomic bins 

Relevant parameters of `pgxLoader` function: type, output, filters, codematches, skip,limit

```{r}
cnv.status <- pgxLoader(type="variant", output="pgxmatrix", filters = "NCIT:C3059")
```

By default, it returns the first 2000 profiles, because loading too many profiles will cause
time-out problem. It is noted that the number of analysis profiles does not necessarily equal the number of samples.

The data looks like this

```{r}
print(dim(cnv.status))
cnv.status[c(1:3), c(1:5,6213:6215)]
```

In this dataframe, `analysis_id` is the identifier for individual analysis, `biosample_id` is the identifier for
individual biosample. One biosample_id may correspond to multiple analysis_id. `group_id` equals the meaning of `filters`. It's followed by all “gain status” columns (3106 intervals) plus all “loss status” columns (3106 intervals). The status is indicated by a coverage value, i.e. the fraction of how much the binned interval overlaps with one or more CNVs of the given type (DUP/DEL). For example, if the column `X1.400000.1400000.DUP` is 1.0 in one row, it means that in the corresponding analysis one or more duplication events occur in the entire (100%) bin located in chromosome 1: 400000-1400000.

If you want to access the next 1000 profiles in addition to the first 2000 analysis profiles, you can query like this

```{r}
cnv.status.2 <- pgxLoader(type="variant", output="pgxmatrix", filters = "NCIT:C3059",
                          skip=2, limit=1000)
```

Setting `codematches = True` can exclude profiles with group_id belonging to child terms of the input filters.

```{r}
cnv.status <- pgxLoader(type="variant", output="pgxmatrix", filters = "NCIT:C3059",
                        codematches = T)
```

Only 46 from the 2000 profiles are left.

```{r}
dim(cnv.status)
```

## Across chromosomes or the whole genome 

Relevant parameters of `pgxLoader` function: type, output, filters, codematches, skip, limit

```{r}
cnv_covergae <- pgxLoader(type="variant", output="coverage", filters = "NCIT:C2948")
```

It includes CNV coverage across chromosome arms, whole chromosomes, or whole genome.

```{r}
names(cnv_covergae)
```

By default, it also returns the first 2000 profiles. You can use parameters `skip` and `limit` to query additional profiles if the group contains more than 2000 profiles.

The data of CNV coverage across chromosomal arms looks like this

```{r}
dim(cnv_covergae$chrom_arm_coverage)
```

The row names are id of biosamples from the group NCIT:C2948. There are 96 columns. 
The first 48 columns are duplication coverage across chromosomal arms, followed by deletion coverage.
The data of CNV coverage across whole chromosomes is similar, with the only difference in columns.

```{r}
head(cnv_covergae$chrom_arm_coverage)[,c(1:4, 49:52)]
```

The data of CNV coverage across genome (hg38) looks like this

```{r}
head(cnv_covergae$whole_genome_coverage)
```

The first column is the total called coverage, followed by duplication coverage and deletion coverage.

Setting `codematches = True` can exclude profiles with group_id belonging to child terms of the input filters.
21 samples are excluded from the original 47 samples in this case.

```{r}
cnv_covergae_2 <- pgxLoader(type="variant", output="coverage", filters = "NCIT:C2948",
                            codematches = T)

cat(paste('original data dimention:', paste(dim(cnv_covergae$chrom_arm_coverage),collapse = '*'),'\ncurrent data dimention:', paste(dim(cnv_covergae_2$chrom_arm_coverage),collapse = '*')))
```

# Retrieve segment variants by biosampleid

Relevant parameters of `pgxLoader` function: type, output, biosample_id, save_file, filename

Because of the time-out problem, the segment variant data can only be accessed by 
individual biosample id instead of filters.

## Get biosamples id from previously retrieved biosample information

```{r}
biosample_id <- biosamples$biosample_id[c(1:20)]
```

There are three output formats. 

## The first output format (by default)

 This format contains variant id and associated biosample id as well as analysis id. The variant is represented as 'DUP' (duplication) or 'DEL' (deletion) in specific chromosome locations. A few variants are point mutations represented by columns `reference_bases` and `alternate_bases`.

```{r}
variant_1 <- pgxLoader(type="variant", biosample_id = biosample_id)
head(variant_1)
```

## The second output format (`output` = "pgxseg")

This format is '.pgxseg' file format. It contains segment mean values (in `log2` column), which are equal to log2(copy number of measured sample/copy number of control sample (usually 2)). 

```{r}
variant_2 <- pgxLoader(type="variant", biosample_id = biosample_id,output = "pgxseg")
head(variant_2)
```

## The third output format (`output` = "seg")

This format is similar to the general '.seg' file format and compatible with IGV tool for visualization. The only difference between this file format and the general '.seg' file format is the fifth column. It represents variant type in this format while in the general '.seg' file format, it represents number of probes or bins covered by the segment. In addition, the point mutation variants are excluded in this file format.

```{r}
variant_3 <- pgxLoader(type="variant", biosample_id = biosample_id,output = "seg")
head(variant_3)
```

# Export variants data for visualization

Setting `save_file` as TRUE in `pgxLoader` function would make this function doesn't return 
variants data directly but let the retrieved data saved in the current work directory by default or other 
paths (specified by `filename`). The export is only available for variants data (type='variant'). 

## Upload 'pgxseg' file to Progenetix website

The following command creates a '.pgxseg' file with the name "variants.pgxseg" in "~/Downloads/" folder. 

```{r}
pgxLoader(type='variant', output='pgxseg', biosample_id=biosample_id, save_file=TRUE, 
          filename="~/Downloads/variants.pgxseg")
```

You can upload this '.pgxseg' file to [this link](https://progenetix.org/service-collection/uploader) 
for visualization.

## Upload '.seg' file to IGV

The following command creates a special '.seg' file with the name "variants.seg" in "~/Downloads/" folder. 

```{r}
pgxLoader(type='variant', output='seg', biosample_id=biosample_id, save_file=TRUE, 
          filename="~/Downloads/variants.seg")
```

You can upload this '.seg' file to [IGV tool](https://software.broadinstitute.org/software/igv/) for visualization.

# Session Info

```{r echo = FALSE}
sessionInfo()
```
